rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake

```diff
@@ -212,9 +212,10 @@ foreach(_typesupport_impl ${_typesupport_impls})
 
   set(_extension_compile_flags "")
   set(_PYTHON_EXECUTABLE ${PYTHON_EXECUTABLE})
-  if(NOT WIN32)
-    set(_extension_compile_flags "-Wall -Wextra")
-  elseif("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
+  if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    set(_extension_compile_flags "-Wall;-Wextra")
```

Comment:
I would suggest to not use a semicolon here. Just pass both value parts as separate values and the variable will be a list (same as the `add_compile_options` invocation above).